### PR TITLE
merge request for searching functionality

### PR DESCRIPTION
**Pull request by rubichandrap**
- fixed PostCard Component not centered at media <430px
- added search functionality
- added active class to nav links when router path is equal with the mapped routes
- added hooks useDebounce

### CODE REVIEW

File: hooks/useDebounce.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [value, delay]);
+
+  return debouncedValue;
+};
+
+export default useDebounce;
```
Comment (elianiva) [inline hooks/useDebounce.ts:18]: prefer named export kalo helper/hooks/sejenisnya

File: hooks/index.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export { default as useDebounce } from './useDebounce';
```
Comment (elianiva) [inline hooks/index.ts:1]: ditto

File: components/Navbar.tsx [hunk 3]
```diff
@@ -1,15 +1,29 @@
 import Link from 'next/link';
 import GithubIcon from '#components/Icons/GithubIcon';
 import TelegramIcon from '#components/Icons/TelegramIcon';
+import React from 'react';
```
Comment (elianiva) [inline components/Navbar.tsx:4]: ini buat apa? bukannya udah otomatis ditambahin sama nextjs?

File: pages/search.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import { useDebounce } from '#hooks/index';
+import React, { useEffect, useState } from 'react';
```
Comment (elianiva) [inline pages/search.tsx:2]: ditto

File: pages/search.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+  const debouncedKeywords = useDebounce<typeof keywords>(keywords);
+
+  useEffect(() => {
+    const getPosts = async () => {
+      const res = await fetch(`/api/search?keywords=${debouncedKeywords}`);
+      const json = await res.json();
+      setPosts(json.data);
+
+      console.log(json.data);
```
Comment (elianiva) [inline pages/search.tsx:18]: lupa belum dihapus om?

File: pages/search.tsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    const { value } = e.target;
+    setKeywords(value);
+  };
+  return (
+    <>
+      <style jsx>{`
+        .thin-border-b {
+          border-bottom-width: 1px;
```
Comment (elianiva) [inline pages/search.tsx:37]: ada alasan tertentu ngga pake `border-b-1` punya tailwind?

**Review by elianiva — CHANGES_REQUESTED** [review]

File: components/Navbar.tsx (hunk 3 above)
Comment (rubichandrap) [reply components/Navbar.tsx:4]: Itu dipake buat typing, React.KeyboardEvent

File: hooks/useDebounce.ts (hunk 1 above)
Comment (rubichandrap) [reply hooks/useDebounce.ts:18]: Udah saya bikin aggregatornya, tp kalo ttp named ya gpp

File: pages/search.tsx (hunk 5 above)
Comment (rubichandrap) [reply pages/search.tsx:18]: Lupa

File: pages/search.tsx (hunk 6 above)
Comment (rubichandrap) [reply pages/search.tsx:37]: Ga ada autocompletenya tadi, jadi saya pikir ga ada

**Review by elianiva — APPROVED** [review]
Thanks!

image

**Review by aldy505 — APPROVED** [review]
I don't have any idea what the code does. But it looks good to me.

**Comment by aldy505** [comment]
laode approved via telegram. let's merge this.

**Pull request merged**